app/controllers/interest_controller.rb

```diff
@@ -25,100 +25,133 @@ class InterestController < ApplicationController
   def list_interests
     store_location
     @title = :list_interests_title.t
-    # notifications = Notification.find_all_by_user_id(@user.id).sort do |a,b|
-    notifications = Notification.where(user_id: @user.id).sort do |a, b|
+    notifications = find_relevant_notifications
+    interests = find_relevant_interests
+    @targets = notifications + interests
+
+    @pages = paginate_numbers(:page, 50)
+    @pages.num_total = @targets.length
+    @targets = @targets[@pages.from..@pages.to]
+  end
+
+  private
+
+  def find_relevant_notifications
+    Notification.for_user(@user).sort do |a, b|
       result = a.flavor.to_s <=> b.flavor.to_s
       result = a.summary.to_s <=> b.summary.to_s if result.zero?
       result
     end
-    # interests = Interest.find_all_by_user_id(@user.id).sort do |a,b|
-    interests = Interest.where(user_id: @user.id).sort do |a, b|
+  end
+
+  def find_relevant_interests
+    Interest.for_user(@user).sort do |a, b|
       result = a.target_type <=> b.target_type
       if result.zero?
         result = (a.target ? a.target.text_name : "") <=>
                  (b.target ? b.target.text_name : "")
       end
       result
     end
-    @targets = notifications + interests
-    @pages = paginate_numbers(:page, 50)
-    @pages.num_total = @targets.length
-    @targets = @targets[@pages.from..@pages.to]
   end
 
+  public
+
   # Callback to change interest state in an object.
   # Linked from: show_<object> and emails
   # Redirects back (falls back on show_<object>)
   # Inputs: params[:type], params[:id], params[:state], params[:user]
   # Outputs: none
-  def set_interest
+  def set_interest # rubocop:disable Metrics/AbcSize
     pass_query_params
-    type   = params[:type].to_s
-    oid    = params[:id].to_i
-    state  = params[:state].to_i
-    uid    = params[:user]
-    target = Comment.find_object(type, oid)
-    if @user
-      interest = Interest.find_by(
-        target_type: type, target_id: oid, user_id: @user.id
+    target_type = params[:type].to_s
+    target_id   = params[:id].to_i
+    state       = params[:state].to_i
+    user_id     = params[:user]
+    target      = Comment.find_object(target_type, target_id)
```

Comment:
Is there a more direct way to get the `target` object than using a `Comment` class method?

I get it, it works, but `target_type.to_s.camelize.constantize.find(target_id)` is simpler.

Reply:
Oh, find_object is from AbstractModel and does some error handling. Changing it back to
`Interest.find_object(target_type, target_id)`

Reply:
"Interest" makes more sense than "Comment".  I admit I did a double-take on that.  I'm sure there is a perfectly sensible historical reason for that ... maybe the method was moved from Comment into AbstractModel after writing this?  In any case, good to see it moved to something more reasonable.  Wait, hell, why not "AbstractModel.find_object"??  That would really emphasize that this has nothing to do with Interest per se, and that we're just looking up an object by type and id!